### PR TITLE
Automated cherry pick of #71560: Don't log a warning to override hostname if there's no

### PR DESCRIPTION
**Pull request by ReillyTevera**
Cherry pick of #71560 on release-1.13.

#71560: Don't log a warning to override hostname if there's no

### CODE REVIEW

**Review by mattjmcnaughton — APPROVED** [review]
/ok-to-test
/lgtm

**Comment by ReillyProcentive** [comment]
Is there anything I need to do to help this along?

**Comment by hoegaarden** [comment]
> Is there anything I need to do to help this along?

The PR is missing the approve label, I want to see if the approver of the original PR or someone in the repective OWNERS file is willing to approve this cherrypick. As soon as that happens and there are no other concerns (which I don't think there are currently) we (the @kubernetes/patch-release-team) will approve this PR for cherrypicking and it will go in into the next patch release.

**Review by KashifSaadat — APPROVED** [review]
/approve

**Comment by cblecker** [comment]
/hold

This is a cleanup, and not a critical bug fix. While the impact is low, this doesn't meet our criteria for a [critical bug fix](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/release/versioning.md#patch-releases).

I'm recommending closure.

**Comment by ReillyProcentive** [comment]
While it is not a major issue it does fall under "affecting a large number of users" as at minimum it affects every single kops user who is running the current supported k8s release (1.12) ( and kops for k8s 1.14 is probably 3-5 months out depending on whether or not they accelerate their release schedule to catch up). 

The issue results in every single kubelet logging a warning message every ~10 seconds which can make it easy for actual problems to get lost in the noise as well as cause problems for those of us who are using log aggregation for centralizing all host logs.

That said, I am not opposed to closing these PRs as we have implemented a filter at the log aggregation layer to exclude these specific messages and would just need to leave that in place until kops gets to 1.14. I defer to the patch-release-team on this one.

**Comment by dashpole** [comment]
I agree this does not warrant a cherrypick.

**Pull request closed without merging**
